src/storage/lock_manager/deadlock.rs

```diff
@@ -198,20 +195,25 @@ impl Inner {
     ) -> Self {
         Self {
             store_id,
-            leader_id: 0,
+            leader_info: None,
+            leader_client: None,
             detect_table: Rc::new(RefCell::new(DetectTable::default())),
             waiter_mgr_scheduler,
             security_mgr,
-            addrs: Rc::new(RefCell::new(HashMap::new())),
-            leader_client: None,
-            recving: Rc::new(RefCell::new(HashMap::new())),
-            failed: Rc::new(RefCell::new(HashSet::new())),
             max_ts: 0,
         }
     }
 
+    fn leader_id(&self) -> u64 {
+        self.leader_info.as_ref().unwrap().0
```

Comment:
How to guarantee that leader_info is Some(xx)?

Reply:
Only `is_leader()` and `reconnect_leader()` call these functions. `is_leader()` has checked `leader_info.is_some()` and L410 has checked it too.